src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,43 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners;
+    private final ExecutorService executorService;
 
+    public EventManager() {
+        this.listeners = new CopyOnWriteArrayList<>();
+        this.executorService = Executors.newCachedThreadPool();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
+    }
 
+    public void deregisterListener(EventListener listener) {
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executorService.shutdown();
+        try {
+            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+                executorService.shutdownNow();
+            }
+        } catch (InterruptedException ex) {
+            executorService.shutdownNow();
```

Comment:
When catching InterruptedException, it's a common practice to restore the interrupt status after calling shutdownNow(), as you did in line 40. However, it's also recommended to log the exception to understand the interruption's context.